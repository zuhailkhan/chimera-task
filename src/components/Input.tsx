import React from 'react'


export interface IProps {
  title: string;
  type: string;
  placeholder?: string;
  name?: string;
  handleChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value: string,
}

const Input = ({title, type, placeholder, name, handleChange, value}: IProps) => {

  return (
    <div className="flex justify-between" title={title + '_container'}>
      <label htmlFor={title}>{title}</label>
      <input
        name={name}
        type={type}
        placeholder={placeholder}
        title={title}
        id={type + title}
        onChange={handleChange}
        value={value}
      />
    </div>
  );
}

export default Input