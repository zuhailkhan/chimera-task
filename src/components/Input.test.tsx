import { cleanup, screen, render } from "@testing-library/react";
import Input, { IProps } from "./Input";

describe("Input Custom Component", () => {
  const defaultProps: IProps = {
    title: "Name",
    type: "text",
    value: "",
    handleChange: jest.fn(),
  };

  afterEach(cleanup)

  it('should render the component', () => {
    const { container } = render(<Input {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  it('should have correct values', () => {
    render(<Input {...defaultProps} value="Zuhail" />)
    const input = screen.getByTitle(defaultProps.title)! as HTMLInputElement
    expect(input.value).toBe('Zuhail')
  })
});
