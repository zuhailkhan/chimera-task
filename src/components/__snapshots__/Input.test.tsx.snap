// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Custom Component should render the component 1`] = `
<div>
  <div
    class="flex justify-between"
    title="Name_container"
  >
    <label
      for="Name"
    >
      Name
    </label>
    <input
      id="textName"
      title="Name"
      type="text"
      value=""
    />
  </div>
</div>
`;
